HISTORY.md

```diff
@@ -1,3 +1,127 @@
+# TileDB v2.7.0 Release Notes
+
+## Disk Format
+
+* Removed file `__lock.tdb` from the array folder and updated the format spec. Also removed config `vfs.file.enable_filelocks`. [#2692](https://github.com/TileDB-Inc/TileDB/pull/2692)
+
+## New features
+
+* Convert deserialize function of Attribute to factory function [#2566](https://github.com/TileDB-Inc/TileDB/pull/2566)
```

Comment:
Not sure how much manual editing we want to do here, but IMO 1 and 3 should not be under New Features because they are internal refactoring.